### PR TITLE
0.14.1 release notes

### PR DESCRIPTION
**Pull request by samredai**

### CODE REVIEW

File: landing-page/content/common/release-notes.md [hunk 1]
```diff
@@ -69,6 +69,31 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
...
 
+## 0.14.1 release
+
+This release includes all bug fixes from the 0.14.x patch releases.
+
+### Notable bug fixes
+
+* Core
+  - API: Fix ID assignment in schema merging ([#5395](https://github.com/apache/iceberg/pull/5395))
+  - AWS: S3OutputStream - failure to close should persist on subsequent close calls ([#5311](https://github.com/apache/iceberg/pull/5311))
```
Comment (Fokko) [inline landing-page/content/common/release-notes.md:80]: Should we put this under Vendor integrations?

File: landing-page/content/common/release-notes.md [hunk 2]
```diff
@@ -69,6 +69,31 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
...
+  - Core: Fix snapshot log with intermediate transaction snapshots ([#5568](https://github.com/apache/iceberg/pull/5568))
+  - Core: Fix exception handling in BaseTaskWriter ([#5683](https://github.com/apache/iceberg/pull/5683))
+  - Core: Support deleting tables without metadata files ([#5510](https://github.com/apache/iceberg/pull/5510))
+  - Core, AWS: Fix Kryo serialization failure for FileIO ([#5437](https://github.com/apache/iceberg/pull/5437))
+  - Core: Add CommitStateUnknownException handling to REST ([#5694](https://github.com/apache/iceberg/pull/5694))
+  - AWS: Ignore flaky test; backport from ([#5379](https://github.com/apache/iceberg/pull/5379))
```
Comment (Fokko) [inline landing-page/content/common/release-notes.md:86]: I think we can just remove this one; I don't think it is notable

File: landing-page/content/common/release-notes.md [hunk 3]
```diff
@@ -69,6 +69,31 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
...
+  - Build: Exclude unnecessary git properties from iceberg-build.properties ([#5277](https://github.com/apache/iceberg/pull/5277))
+  - Build: Fix Scala 2.13 builds in stage-binaries.sh. ([#5270](https://github.com/apache/iceberg/pull/5270))
+  - Build: Add iceberg-build.properties to RAT excludes. ([#5262](https://github.com/apache/iceberg/pull/5262))
```
Comment (Fokko) [inline landing-page/content/common/release-notes.md:89]: I think we can just leave out the Build ones 👍🏻 They aren't very notable

File: landing-page/content/common/release-notes.md [hunk 4]
```diff
@@ -69,6 +69,31 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
...
 
 
+## 0.14.1 release
+
+This release includes all bug fixes from the 0.14.x patch releases.
+
+### Notable bug fixes
+
+* Core
+  - API: Fix ID assignment in schema merging ([#5395](https://github.com/apache/iceberg/pull/5395))
```
Comment (Fokko) [inline landing-page/content/common/release-notes.md:79]: Let's split out this into API to be consistent with the other release notes

**Review by Fokko — COMMENTED** [review]
Thanks for picking this up @samredai, few small nits

**Comment by samredai** [comment]
Updated based on your comments, thanks @Fokko!

File: landing-page/content/common/release-notes.md [hunk 5]
```diff
@@ -69,6 +69,29 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
...
+  - Spark: Fix stats in rewrite metadata action ([#5691](https://github.com/apache/iceberg/pull/5691))
+* File Formats
+  - Parquet: Close zstd input stream early to avoid memory pressure ([#5681](https://github.com/apache/iceberg/pull/5681))
+* Vendor Integrations
+  - Core, AWS: Fix Kryo serialization failure for FileIO ([#5437](https://github.com/apache/iceberg/pull/5437))
+  - AWS: S3OutputStream - failure to close should persist on subsequent close calls ([#5311](https://github.com/apache/iceberg/pull/5311))
+
+## Past releases
+
 ## 0.14.0 release
```
Comment (Fokko) [inline landing-page/content/common/release-notes.md:95]:
```suggestion
### 0.14.0 release
```

**Review by Fokko — APPROVED** [review]
Thanks for picking this up @samredai! Much appreciated 👍🏻

**Pull request merged**